setup.py

```diff
@@ -28,7 +28,7 @@
     jax_version = [line[4:].strip() for line in f.readlines() if "jax=" in line][0]
 
 requirements = [
-    "pennylane>=0.32",
+    "pennylane @ git+https://github.com/PennyLaneAI/pennylane.git",
```

Comment:
@rmoyard will this need to be reverted at some point?

Reply:
This definitely needs to be updated to `pennylane>=0.33`, the only question is when 🤔 We could wait merging the PR until that release becomes available, or merge now with this and update it in a new PR once the release is out.

Reply:
Yes will be updated to pennylane 0.33, I don't have strong preferences both options are good on my side.

Reply:
Tom wants the plugin matrix to pass before the release, so we'll have to merge this in now :)

Reply:
Ok no worries I am updating amazon plugin in the CI to pass the tests

Reply:
Something to keep in mind, but whenever you update `check-catalyst.yaml`, chances are you also need to update `check-pl-compat.yaml` (we'll have to see if we can improve that in the future).